arkouda/client.py

```diff
@@ -523,9 +535,71 @@ def shutdown() -> None:
     serverConfig = None
 
 
+def _json_args_to_str(json_obj: Dict) -> Tuple[int, str]:
+    """
+    Convert Python Dictionary into a JSON formatted string that can be parsed by the msg
+    processing system on the Arkouda Server
+
+    Parameters
+    ----------
+    json_obj : dict
+        Python dictionary of key:val representing command arguments
+
+    Return
+    ------
+    Tuple - the number of parameters found and the json formatted string
+
+    Raises
+    ------
+    TypeError
+        - Keys are a type other than str
+        - Any value is a dictionary.
+        - A list contains values of multiple types.
+
+    Notes
+    -----
+    - Nested dictionaries are not yet supported, but are planned for future support.
+    - Support for lists of pdarray or Strings objects does not yet exist.
+    """
+    from arkouda.pdarrayclass import pdarray
+    from arkouda.strings import Strings
+
+    j = []
+    for key, val in json_obj.items():
+        if not isinstance(key, str):
+            raise TypeError(f"Argument keys are required to be str. Found {type(key)}")
+        if isinstance(val, dict):
+            raise TypeError("Nested JSON is not currently supported for server messages.")
+        param = {"key": key}
+
+        if isinstance(val, pdarray):
+            param["objType"] = "pdarray"
+            param["dtype"] = str(val.dtype)
+            param["val"] = val.name
+        elif isinstance(val, Strings):
+            param["objType"] = "SegString"
+            param["dtype"] = "str"
+            # empty string if name of String obj is none
+            param["val"] = val.name if val.name else ""
+        elif isinstance(val, list):
+            param["objType"] = "list"
+            dtypes = set([p.dtype for p in val])
+            if len(dtypes) > 1:
+                t_str = ", ".join(dtypes)
+                raise TypeError(f"List values must be of the same type. Found {t_str}")
```

Comment:
@pierce314159 10-4, I thought about this as well, but initially we do not seem to require it. Added an issue for tracking in the event that we need it sooner rather than later.

#1632